### PR TITLE
[runtime-security] fix dentry resolution from lower layer for overlayfs 

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This fixes marshaling issue when having `"` in filename. It also remove the dedicated mount perf as it generates a race between mount and file event.

Ex: file access right after a mount
As the mount event can be consumed after the file event, the mount resolution will fail.

### Motivation

Avoid rules not matching due to mount resolution failure

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

`DentryOverlay` functionnal tests trigger the issue as they mount `overlayfs` filesystem

### CODE REVIEW

**Review by lebauce — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
lgtm :)

File: pkg/security/probe/dentry_resolver.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 
 	"github.com/DataDog/datadog-agent/pkg/security/ebpf"
 )
+import "github.com/cobaugh/osrelease"
```
Comment (lebauce) [inline pkg/security/probe/dentry_resolver.go:21]: nit: could be moved inside the previous import section

**Review by Gui774ume — APPROVED** [review]
<3

**Review by guedou — APPROVED** [review]
Awesome PR! My custom overlayfs tess all pass.

**Pull request closed without merging**